File: Python/binary_search.py
def binary_search(arr: list[int], needle: int):
    left = 0
    right = len(arr) - 1
    while left < right:
        m = right + left // 2
        if arr[m] == needle:
            return True
        if arr[m] > needle:
            right = m
        else:
            left = m + 1
    return False


print(binary_search([169, 27, 118, 49, 14, 205, 448, 303], 303))
print(binary_search([169, 27, 118, 49, 14, 205, 448, 303], 3033))

File: JS/binary_search.js
function bs(arr, needle) {
  let h = arr.length;
  let l = 0;
  while (l < h) {
    let m = Math.floor(l + (h - l) / 2);
    const v = arr[m];
    if (v === needle) {
      return true;
    } else if (v > needle) {
      h = m;
    } else {
      l = m + 1;
    }
  }
  return false;
}

console.log(bs([1, 5, 7, 11, 15, 22, 30], 7));
console.log(bs([1, 5, 7, 11, 15, 22, 30], 30));
console.log(bs([1, 5, 7, 11, 15, 22, 30], 33));
